WordPress/Classes/Services/NullBlogPropertySanitizer.swift

```diff
@@ -58,6 +58,13 @@ import Foundation
 
             try? self.context.save()
         }
+
+        // Make sure the "sanitization" work is done before any other Core Data reads or writes.
+        if Thread.isMainThread, context.concurrencyType == .mainQueueConcurrencyType {
```

Comment:
From my understanding, this check isn't really necessary and more like a "double check", right?

Also... not knowing the codebase and standards very well, I see this class is out of the `ContextManager` scheme and I wonder if would make sense to somehow add this routine to the queues managed there (which of course would make it harder for it to be before other reads or writes).

Reply:
> From my understanding, this check isn't really necessary and more like a "double check", right?

In most cases, probably yes. Because the `ContextManager.shared` singleton is created from a main thread which is the only place that calls this sanitizer function. But there is no guarantee that's always going to be the case. Unless we add a `precondition(Thread.isMainThread)` to say it's a programming error to call it in a background thread. BTW, I believe some unit tests call this function from a background thread.

> [...] add this routine to the queues managed there

My understanding of this sanitizer function is, it's supposed to be call as the very first Core Data operation, to ensure there is no models that have a blog property that's null. If that's indeed the case, we'll want to call this function immediately after the Core Data stack is initialized, to eliminated the race condition between this sanitizing work and other parts of app's Core Data calls. Calling it synchronously would has a better chance in achieving that than sending the function to a queue.